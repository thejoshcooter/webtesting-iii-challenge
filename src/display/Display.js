import React from 'react';

const Display = ({ closed, locked }) => {
  const closedClass = `led ${closed ? 'red-led' : 'green-led'}`;
  const lockedClass = `led ${locked ? 'red-led' : 'green-led'}`;

  return (
    <div className="display panel">
      <div className={lockedClass} data-testid='lockUnlock'>{locked ? 'Locked' : 'Unlocked'}</div>
      <div className={closedClass} data-testid='openClose'>{closed ? 'Closed' : 'Open'}</div>
    </div>
  );
};

Display.defaultProps = {
  closed: false,
  locked: false,
};

export default Display;
