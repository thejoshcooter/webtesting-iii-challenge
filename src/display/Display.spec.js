import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import 'jest-dom/extend-expect';
import '@testing-library/react/cleanup-after-each';

import Display from './Display';

describe('<Display />', () => {
    it('checks snapshot', () => {
        const tree = renderer.create(<Display />).toJSON();
        expect(tree).toMatchSnapshot();
    })
    it('shows open/unlocked', () => {
        const { getByText } = render(<Display />);
        getByText(/unlocked/i);
        getByText(/open/i);
    })
    it('can close', () => {
        const { getByTestId } = render(<Display closed={true} />);
        const open = getByTestId('openClose');
        expect(open).toHaveTextContent(/closed/i);
    })
    it('can lock', () => {
        const { getByTestId } = render(<Display locked={true} />);
        const lock = getByTestId('lockUnlock');
        expect(lock).toHaveTextContent(/locked/i);
    })

    it('styles unlocked/opeen', () => {
        const { getByTestId } = render(<Display locked={false} closed={false} />);
        const lock = getByTestId('lockUnlock');
        const open = getByTestId('openClose');
        expect(lock).toHaveClass('green-led');
        expect(open).toHaveClass('green-led');
    })

    it('styles locked/closed', () => {
        const { getByTestId } = render(<Display locked={true} closed={true} />);
        const lock = getByTestId('lockUnlock');
        const open = getByTestId('openClose');
        expect(lock).toHaveClass('red-led');
        expect(open).toHaveClass('red-led');
    })
    
})