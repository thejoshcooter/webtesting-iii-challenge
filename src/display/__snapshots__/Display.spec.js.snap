// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display /> checks snapshot 1`] = `
<div
  className="display panel"
>
  <div
    className="led green-led"
    data-testid="lockUnlock"
  >
    Unlocked
  </div>
  <div
    className="led green-led"
    data-testid="openClose"
  >
    Open
  </div>
</div>
`;
